src/main/java/org/openmrs/addonindex/domain/artifactory/ArtifactoryFolderInfoChild.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.addonindex.domain.artifactory;
+
+import org.openmrs.addonindex.util.Version;
+
+public class ArtifactoryFolderInfoChild implements Comparable<ArtifactoryFolderInfoChild> {
+    private String uri;
+    private boolean folder;
+
+    public String getUri() {
+        return uri;
+    }
+
+    public void setUri(String input) {
+        this.uri = input;
+    }
+
+    public boolean getFolder() {
+        return folder;
+    }
+
+    public void setFolder(boolean input) {
+        this.folder = input;
+    }
+
+    private Version version;
+
+    public Version getVersion() {
+        return version;
+    }
+
+    public void setVersion(Version version) {
+        this.version = version;
+    }
+
+    @Override
+    public int compareTo(ArtifactoryFolderInfoChild o) {
+        return 0;
```

Comment:
This is wrong. You should only make something implement Comparable if you are actually going to implement its compareTo method. In this case I don't see why these would be comparable since they don't have a natural ordering.

So, get rid of the compareTo method and remove the "implements Comparable"